### PR TITLE
Decimal to binary

### PR DESCRIPTION
**Pull request by HebaShaheen**
---
name: solution review 
about: A template PR for code review with a checklist
---

### Files

- [x] The file name describes the function's behavior
- [x] There is a module docstring in the function file
- [x] The test file's name matches the function file name -
  `/tests/test_file_name.py`
- [x] There is a module docstring in the tests file

### Unit Tests

- [x] The test class has a helpful name in PascalCase
- [x] The test class has a docstring
- Every unit test has
  - [x] A helpful name
  - [ ] A clear docstring
  - [x] Only one assertion
  - [x] There is no logic in the unit test
- [x] All tests pass
- [x] There are tests for defensive assertions
- [x] There are tests for boundary cases

### Function Docstring

- [x] The function's behavior is described
- The function's arguments are described:
  - [x] Type
  - [x] Purpose
  - [ ] Other assumptions (eg. if it's a number, what's the expected range?)
- The return value is described
  - [x] Type
  - [ ] Other assumptions are documented
- The defensive assertions are documented using `Raises:`
  - [x] Each assumption about an argument is checked with an assertion
  - [ ] Each assertion checks for _only one_ assumption about the argument
- [x] Include 3 or more (passing!) doctests

### The Function

- [x] The function's name describes it's behavior
- [x] The function's name matches the file name
- [x] The function has correct type annotations

### Do's

- [x] Variable names help to understand the strategy
- [ ] Any comments are clear and describe the strategy
- [ ] Lines of code are spaced to help show different stages of the strategy

### Don'ts

- [x] The function's strategy _is not_ described in the documentation
- [x] Comments explain the _strategy_, **not** the _implementation_
- [x] The function _does not_ have more comments than code
  - If it does, consider finding a new strategy or a simpler implementation

## Implementation

- [x] The code passes the formatting checks
- [x] The code passes all Ruff linting checks
- [x] The code has no (reasonable) Pylint errors
  - In code review, you can decide when fixing a Pylint error is helpful and
    when it's too restricting.
- [ ] Variables are named with snake_case
- [ ] Variable names are clear and helpful
- [x] The code follows the strategy as simply as possible
- [x] The implementation is as simple as possible given the strategy
- [x] The code includes defensive assertions
- [ ] Defensive assertions include as little logic as possible

### CODE REVIEW

File: solutions/tests/test_decimal_to_binary.py [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+@author: Heba Shaheen
+"""
+
+import unittest
+
+from ..decimal_to_binary import decimal_to_binary
+
+
+class TestBinary(unittest.TestCase):
```
Comment (sashour82) [inline solutions/tests/test_decimal_to_binary.py:14]: Name the test class more helpful name (ie: TestDecimalToBinary)

File: solutions/tests/test_decimal_to_binary.py [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+        actual = decimal_to_binary(15)
+        expected = "1111"
+        self.assertEqual(actual, expected)
+
+    def test_non_integer_input(self):
+        """It should raise an assertion error if the argument is not an integer"""
+        with self.assertRaises(AssertionError):
+            decimal_to_binary("1")
+
+    def test_non_negative_input(self):
```
Comment (sashour82) [inline solutions/tests/test_decimal_to_binary.py:46]: It should be (test_negative_input) as name of the unitest

File: solutions/tests/test_decimal_to_binary.py [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+import unittest
+
+from ..decimal_to_binary import decimal_to_binary
+
+
+class TestBinary(unittest.TestCase):
+    """Unit tests for decimal_to_binary function."""
+
+    def test_number_0(self):
+        """It should give the binary"""
```
Comment (sashour82) [inline solutions/tests/test_decimal_to_binary.py:18]: All unit test docstring should be more specific (i.e: It should return xxx as binary)

File: solutions/decimal_to_binary.py [hunk 4]
```diff
@@ -0,0 +1,41 @@
...
+# -*- coding: utf-8 -*-
+"""
+A function to convert decimal numbers to binary.
+
+Created on 1/1/2025
+@author: Heba Shaheen
+"""
+
+
+def decimal_to_binary(n: int) -> str:
```
Comment (sashour82) [inline solutions/decimal_to_binary.py:11]: variable n should be clear to align with function behavior (i.e: decimal_input)

**Comment by sashour82** [comment]
Good job..

**Review by sashour82 — APPROVED** [review]
Good Job Heba..
Everything is ok.. I'll merge it (decimal to binary) into main
